Unit Test Failure in Text Classifier on Python 2.7/3.6

The following unit test failed on [this](https://travis-ci.com/apple/turicreate/jobs/171305938) build: 
```
________________ TextClassifierCreateTests.test_validation_set _________________
self = <turicreate.test.test_text_classifier.TextClassifierCreateTests testMethod=test_validation_set>
    def test_validation_set(self):
        train = self.data
        valid = self.data
    
        # Test with a validation set
        model = tc.text_classifier.create(train, target='rating', validation_set=valid)
        self.assertTrue('Validation Accuracy' in model.classifier.progress.column_names())
    
        # Test without a validation set
        model = tc.text_classifier.create(train, target='rating', validation_set=None)
        self.assertTrue('Validation Accuracy' not in model.classifier.progress.column_names())
    
        # Test 'auto' validation set
        big_data = train.append(tc.SFrame({
            'rating': [5] * 100,
            'place': ['d'] * 100,
            'text': ['large enough data for %5 percent validation split to activate'] * 100
        }))
        model = tc.text_classifier.create(big_data, target='rating', validation_set='auto')
>       self.assertTrue('Validation Accuracy' in model.classifier.progress.column_names())
E       AssertionError: False is not true
turicreate/test/test_text_classifier.py:182: AssertionError
```

--- a/src/unity/python/turicreate/toolkits/_supervised_learning.py
+++ b/src/unity/python/turicreate/toolkits/_supervised_learning.py
@@ -317,7 +317,7 @@ def create(dataset, target, model_name, features=None,
         if dataset.num_rows() >= 100:
             if verbose:
                 print_validation_track_notification()
-            dataset, validation_set = dataset.random_split(.95, seed=seed)
+            dataset, validation_set = dataset.random_split(.95, seed=seed, exact=True)
         else:
             validation_set = _turicreate.SFrame()
     elif validation_set is None:
@@ -386,7 +386,7 @@ def create_classification_with_model_selector(dataset, target, model_selector,
             if dataset.num_rows() >= 100:
                 if verbose:
                     print_validation_track_notification()
-                dataset, validation_set = dataset.random_split(.95)
+                dataset, validation_set = dataset.random_split(.95, exact=True)
             else:
                 validation_set = None
         else:

--- a/src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py
+++ b/src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py
@@ -177,8 +177,7 @@ def create(input_dataset, target, feature=None, validation_set='auto',
                 if verbose:
                     print ( "PROGRESS: Creating a validation set from 5 percent of training data. This may take a while.\n"
                             "          You can set ``validation_set=None`` to disable validation tracking.\n")
-                dataset, validation_dataset = dataset.random_split(
-                    TRAIN_VALIDATION_SPLIT)
+                dataset, validation_dataset = dataset.random_split(TRAIN_VALIDATION_SPLIT, exact=True)
             else:
                 validation_set = None
                 validation_dataset = _tc.SFrame()

--- a/src/unity/python/turicreate/toolkits/sound_classifier/sound_classifier.py
+++ b/src/unity/python/turicreate/toolkits/sound_classifier/sound_classifier.py
@@ -168,7 +168,7 @@ def create(dataset, target, feature, max_iterations=10,
         if len(dataset) >= 100:
             print ( "Creating a validation set from 5 percent of training data. This may take a while.\n"
                     "\tYou can set ``validation_set=None`` to disable validation tracking.\n")
-            dataset, validation_set = dataset.random_split(0.95)
+            dataset, validation_set = dataset.random_split(0.95, exact=True)
         else:
             validation_set = None
 

